.github/workflows/python-publish.yml

```diff
@@ -6,26 +6,53 @@ name: Upload Python Package
 on:
   release:
     types: [created]
+  workflow_dispatch:
 
 jobs:
-  deploy:
+  deploy-unix:
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install setuptools wheel twine Cython numpy
+          python setup.py install
+      - name: Build and publish
+        env:
+          TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+        run: |
+          python setup.py sdist
+          twine upload --repository testpypi dist/*
```

Comment:
I recommend installing and running tests on a package before uploading it. It has caught some errors that I wouldn't have found otherwise (except by installing the released package from PyPI).

Reply:
For signac, we adopted a convention that branches named `release/X.Y.Z` would automatically trigger builds for testpypi, and tags `vX.Y.Z` trigger builds for pypi. testpypi is always part of [the workflow](https://github.com/glotzerlab/signac/blob/d639e682ca7ebaff781d68621a2d86cc26de04b9/.circleci/config.yml#L149-L169) I use for releases because it catches some issues in the process that might slip by otherwise.

Reply:
Added a test clause right before the deploy.

As for the different branches, I'm not a fan of managing processes through branches.  I was thinking of keeping a workflow that deploys to testpypi but is only triggered through `workflow_dispatch`, so it can be manually tested before making a new tag.

Reply:
That’s cool. Manual triggers for the testpypi job sound even better.